### PR TITLE
nctl debug command line switch

### PR DESCRIPTION
**Pull request by zacshowa**
Added an optional flag to nctl-compile to allow for easier non-release compilation.

### CODE REVIEW

File: utils/nctl/sh/assets/compile.sh
Comment (rafal-ch) [inline utils/nctl/sh/assets/compile.sh]: Is this intentional? For me it results in:
```
Command 'pull' not found, did you mean
```

File: utils/nctl/sh/assets/compile_node_launcher.sh
Comment (rafal-ch) [inline utils/nctl/sh/assets/compile_node_launcher.sh]: It's not a `nctl-compile` it this case, right? It's rather `compile_node_launcher`.

I would suggest either of the two approaches:

1. use the generic `this script only accepts...` and then stop the script execution
2. (preferred by me) just silently ignore unknown parameter and continue

However, I'm not familiar with the requirements, so this is just a suggestion.

File: utils/nctl/sh/assets/compile.sh
Comment (rafal-ch) [inline utils/nctl/sh/assets/compile.sh]: Handling the parameter like that will result in the following:
```
magister@zacisze:~/Casper/casper-node$ nctl-compile-node
Finished release [optimized] target(s) in 7m 03s
magister@zacisze:~/Casper/casper-node$ nctl-compile-node -d
Finished dev [unoptimized + debuginfo] target(s) in 0.12s
magister@zacisze:~/Casper/casper-node$ nctl-compile-node
Finished dev [unoptimized + debuginfo] target(s) in 0.12s
```

So the third time I started it **without** `-d` flag, yet it still compiled the `dev`.

File: utils/nctl/sh/assets/compile_client.sh
Comment (rafal-ch) [inline utils/nctl/sh/assets/compile_client.sh]: Hmm, just thinking out loud. Maybe we should consider being consistent with the approach taken by Rust ecosystem and by default compile in `debug` and support the `--release` command line option?

That might be a too much of a lift, though, since `NCTL` is already wired-up in many places that expect it to compile in release by default.

So treat it just as something for consideration.

CC: @TomVasile

File: utils/nctl/sh/assets/compile.sh
Comment (zacshowa) [reply utils/nctl/sh/assets/compile.sh]: This was not intentional and has been fixed in my local files.

File: utils/nctl/sh/assets/compile_node_launcher.sh
Comment (zacshowa) [reply utils/nctl/sh/assets/compile_node_launcher.sh]: I agree that that this error handling might be a little ambitious for this part of the tool. I chose to implement option 2 and just ignore unknown parameters.

File: utils/nctl/sh/assets/compile.sh
Comment (rafal-ch) [reply utils/nctl/sh/assets/compile.sh]: ~Can you also add this fix to the PR?~

Nvm, I was looking at the wrong commit. All good :+1:

File: utils/nctl/sh/assets/compile.sh
Comment (rafal-ch) [reply utils/nctl/sh/assets/compile.sh]: Works as intended, thanks :+1:

File: utils/nctl/sh/assets/compile_node.sh
Comment (rafal-ch) [inline utils/nctl/sh/assets/compile_node.sh:28]: Can you add a short info to [the docs](https://github.com/casper-network/casper-node/blob/dev/utils/nctl/docs/commands-setup.md?plain=1#L11) on how to switch between compile modes? I think it's worth mentioning that `debug` mode takes preference, i.e., passing `-d` will override the `NCTL_COMPILE_TARGET=release` and vice versa.

**Review by rafal-ch — APPROVED** [review]
LGTM, no blockers :+1: 

But before merging, please get an additional approval from a person with better bash skills than mine :)

File: utils/nctl/sh/assets/compile_node_launcher.sh
Comment (TomVasile) [inline utils/nctl/sh/assets/compile_node_launcher.sh:28]: nit pick: double newline

**Review by TomVasile — APPROVED** [review]
Tested locally and both the new and old method works as expected. Thanks for the PR!

File: utils/nctl/sh/assets/compile_node.sh
Comment (TomVasile) [reply utils/nctl/sh/assets/compile_node.sh:28]: second adding to docs

**Pull request merged**
